cmd/kubelet/app/server_linux.go

```diff
@@ -17,30 +17,38 @@ limitations under the License.
 package app
 
 import (
+	"github.com/fsnotify/fsnotify"
 	"k8s.io/klog/v2"
-	"k8s.io/utils/inotify"
 )
 
 func watchForLockfileContention(path string, done chan struct{}) error {
-	watcher, err := inotify.NewWatcher()
+	watcher, err := fsnotify.NewWatcher()
 	if err != nil {
 		klog.ErrorS(err, "Unable to create watcher for lockfile")
 		return err
 	}
-	if err = watcher.AddWatch(path, inotify.InOpen|inotify.InDeleteSelf); err != nil {
+	if err = watcher.Add(path); err != nil {
 		klog.ErrorS(err, "Unable to watch lockfile")
 		watcher.Close()
 		return err
 	}
 	go func() {
-		select {
-		case ev := <-watcher.Event:
-			klog.InfoS("inotify event", "event", ev)
-		case err = <-watcher.Error:
-			klog.ErrorS(err, "inotify watcher error")
+		defer func() {
+			close(done)
+			watcher.Close()
+		}()
+		for {
```

Comment:
@wzshiming I don't understand why we didn't have the `for` loop before... any idea?
/assign

Reply:
The original `k8s.io/utils/inotify` can choose the registered event when `AddWatch`, but `github.com/fsnotify/fsnotify` can't, so filter after getting.

I missed `inotify.InOpen`, this event there is no watched in `github.com/fsnotify/fsnotify`
No wonder there is only one left `k8s.io/utils/inotify`

The logic here is mainly to exit Kubelet if someone opens or deletes the lockfile
https://github.com/kubernetes/kubernetes/blob/254c77bb8905b80981fa6e4dc5c98c7616f567c0/cmd/kubelet/app/server.go#L531-L543

Could I modify the logic to achieve the same effect?